### PR TITLE
Versioning update

### PR DESCRIPTION
**Pull request by graydon**
This is a set of changes to the env to support the [documented strategy](https://github.com/stellar/stellar-core/blob/master/docs/versioning-soroban.md) for versioning soroban and stellar-core together. Mainly it backs out some partial changes I made earlier that were aiming for a different strategy, as well as includes a change to the interface version to become a pair of protocol version and prerelease version. There'll be another bit of stellar-core-side work to hook this up entirely (as well as a bit of work in the CLI to display the version correctly) but the SDK shouldn't have to change.

### CODE REVIEW

**Review by dmkozh — APPROVED** [review]

File: soroban-env-common/src/meta.rs [hunk 1]
```diff
@@ -5,58 +5,52 @@
 //! [Env](crate::Env) interface provided by the host, rather than a cryptic failure
 //! due to a runtime host function signature mismatch.
 
-// There are two different notions of versioning in this system:
+// Currently the only constant is `INTERFACE_VERSION` which is a u64 with a low
+// and high component. The low component is a pre-release verison which should
```
Comment (sisuresh) [inline soroban-env-common/src/meta.rs:9]: Nit: verison -> version

Comment (graydon) [inline soroban-env-common/src/meta.rs:9]:
```suggestion
// and high component. The low component is a pre-release version which should
```

File: soroban-env-common/src/meta.rs [hunk 2]
```diff
@@ -5,58 +5,52 @@
...
+    pre_release_version: 37,
 );
+
+pub fn get_ledger_protocol_version(interface_version: u64) -> u32 {
+    // The ledger protocol version is the high 32 bits of INTERFACE_VERSION
+    (interface_version >> 32) as u32
+}
+
+pub fn get_pre_release_version(interface_version: u64) -> u32 {
+    // The protocol version is the low 32 bits of INTERFACE_VERSION
```
Comment (graydon) [inline soroban-env-common/src/meta.rs:54]:
```suggestion
    // The pre-release version is the low 32 bits of INTERFACE_VERSION
```

[the remark this answers is not in the document]
Comment (graydon) [reply soroban-env-common/src/meta.rs:54]: oh, sure! Fwiw for typo-style comments like this, the little icon above the comment box with `[+-]` allows you to make an in-line "suggestion" which I can then just push a button to add as an amendment.

**Pull request merged**